apm-agent-core/src/main/java/co/elastic/apm/util/HexUtils.java

```diff
@@ -64,22 +64,34 @@ public static void writeByteAsHex(byte b, StringBuilder sb) {
         sb.append(hexArray[v & 0x0F]);
     }
 
-    public static void writeLongAsHex(long l, JsonWriter jw) {
-        for (int i = 7; i >= 0; i--) {
-            int v = (int) l & 0xFF;
-            jw.writeByte((byte) hexArray[v >>> 4]);
-            jw.writeByte((byte) hexArray[v & 0x0F]);
-            l >>= 8;
+    public static byte getNextByte(String hexEncodedString, int offset) {
+        final int hi = hexCharToBinary(hexEncodedString.charAt(offset));
+        final int lo = hexCharToBinary(hexEncodedString.charAt(offset + 1));
+        if (hi == -1 || lo == -1) {
+            throw new IllegalArgumentException("Not a hex encoded string: " + hexEncodedString + " at offset " + offset);
         }
+        return (byte) ((hi << 4) + lo);
     }
 
-    public static byte getNextByte(String hexEncodedString, int offset) {
-        return (byte) ((Character.digit(hexEncodedString.charAt(offset), 16) << 4)
-            + Character.digit(hexEncodedString.charAt(offset + 1), 16));
+    private static int hexCharToBinary(char ch) {
```

Comment:
why do this instead of `Character.digit(ch, 16)`? is this faster?
couldn't you just do (in getNextByte):

```java
final int hi = Character.digit(hexEncodedString.charAt(offset), 16);
final int lo = Character.digit(hexEncodedString.charAt(offset + 1), 16);
(rest is the same)
```
?

Reply:
TBH, I didn't measure it but I suppose this is faster. Semantically, it's equivalent